phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRepairRegionScanner.java

```diff
@@ -178,6 +183,28 @@ protected void repairIndexRows(Map<byte[], List<Mutation>> indexMutationMap,
         return actualIndexMutationMap;
     }
 
+    private Map<byte[], List<Mutation>> populateActualIndexMutationMap() throws IOException {
+        Map<byte[], List<Mutation>> actualIndexMutationMap = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
+        Scan indexScan = new Scan();
+        indexScan.setTimeRange(scan.getTimeRange().getMin(), scan.getTimeRange().getMax());
+        indexScan.setRaw(true);
+        indexScan.setMaxVersions();
+        indexScan.setCacheBlocks(false);
+        try (RegionScanner regionScanner = region.getScanner(indexScan)) {
```

Comment:
Does the below code work if there is no index row?

Reply:
@gokceni I am not sure I understood your concern here ?

Reply:
If the index is truncated for some reason and this code is run, the below do while code, will it work since you seem to be checking the hasMore in the while part of the loop?

Reply:
@gokceni  hasMore will be false in that case and the code should exit the loop. This is a standard pattern used in multiple places in the index rebuild code.